kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt

```diff
@@ -0,0 +1,374 @@
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.widget.SwitchCompat
+import androidx.constraintlayout.widget.ConstraintLayout
+import com.esri.arcgisruntime.ArcGISRuntimeEnvironment
+import com.esri.arcgisruntime.data.OgcFeatureCollectionTable
+import com.esri.arcgisruntime.data.QueryParameters
+import com.esri.arcgisruntime.data.ServiceFeatureTable
+import com.esri.arcgisruntime.geometry.Geometry
+import com.esri.arcgisruntime.geometry.GeometryEngine
+import com.esri.arcgisruntime.layers.FeatureLayer
+import com.esri.arcgisruntime.loadable.LoadStatus
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.BasemapStyle
+import com.esri.arcgisruntime.mapping.TimeExtent
+import com.esri.arcgisruntime.mapping.view.MapView
+import com.esri.arcgisruntime.sample.querywithcqlfilters.databinding.ActivityMainBinding
+import com.esri.arcgisruntime.sample.querywithcqlfilters.databinding.CqlFiltersLayoutBinding
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol
+import com.esri.arcgisruntime.symbology.SimpleRenderer
+import com.google.android.material.bottomsheet.BottomSheetDialog
+import com.google.android.material.floatingactionbutton.FloatingActionButton
+import java.util.*
+
+
+class MainActivity : AppCompatActivity() {
+
+    // Keep loadable in scope to avoid garbage collection
+    private lateinit var ogcFeatureCollectionTable: OgcFeatureCollectionTable
+
+    // List of CQL where clauses
+    private val cqlQueryList: List<String> = listOf(
+        "F_CODE = 'AP010'",
+        "{ \"eq\" : [ { \"property\" : \"F_CODE\" }, \"AP010\" ] }",
+        "F_CODE LIKE 'AQ%'",
+        "{\"and\":[{\"eq\":[{ \"property\" : \"F_CODE\" }, \"AP010\"]},{ \"before\":" +
+                "[{ \"property\" : \"ZI001_SDV\"},\"2013-01-01\"]}]}",
+        ""
+    )
+
+    // Current selected where query
+    private var cqlQueryListPosition = 4
+
+    // Number of features query should return. Default is 1000
+    private var maxFeatures = 1000
+
+    // When set to true, query searches between fromDate-toDate
+    private var isDateFilter = false
+
+    // Defines date range in queryParameters.timeExtent
+    private var fromDate = Calendar.getInstance()
+    private var toDate = Calendar.getInstance()
+
+    private val activityMainBinding by lazy {
+        ActivityMainBinding.inflate(layoutInflater)
+    }
+
+    private val mapView: MapView by lazy {
+        activityMainBinding.mapView
+    }
+
+    private val fab: FloatingActionButton by lazy {
+        activityMainBinding.fab
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(activityMainBinding.root)
+
+
+        // Authentication with an API key or named user is required to
+        // access basemaps and other location services
+        ArcGISRuntimeEnvironment.setApiKey(BuildConfig.API_KEY)
+
+        // Create a map with the BasemapType topographic
```

Comment:
could use a plain english comment style here

```suggestion
        // Create a map with a topographic basemap
```